### PR TITLE
Support ComplexDataVector in observations

### PR DESCRIPTION
**Pull request by nilsvu**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/NumericalAlgorithms/Interpolation/RegularGridInterpolant.hpp [hunk 1]
```diff
@@ -81,6 +81,9 @@ class RegularGrid {
   void interpolate(gsl::not_null<DataVector*> result,
                    const DataVector& input) const;
   DataVector interpolate(const DataVector& input) const;
+  void interpolate(gsl::not_null<ComplexDataVector*> result,
+                   const ComplexDataVector& input) const;
+  ComplexDataVector interpolate(const ComplexDataVector& input) const;
```
Comment (wthrowe) [inline src/NumericalAlgorithms/Interpolation/RegularGridInterpolant.hpp:86]: Test these.

**Review by wthrowe — CHANGES_REQUESTED** [review]

File: src/NumericalAlgorithms/Interpolation/RegularGridInterpolant.hpp (hunk 1 above)
Comment (nilsvu) [reply src/NumericalAlgorithms/Interpolation/RegularGridInterpolant.hpp:86]: Added a test

**Review by wthrowe — COMMENTED** [review]
Tests look fine.  A lot of the CI jobs are failing, though.

**Review by wthrowe — APPROVED** [review]

File: src/ParallelAlgorithms/Events/ObserveFields.hpp [hunk 2]
```diff
@@ -235,24 +235,44 @@ class ObserveFields<VolumeDim, tmpl::list<Tensors...>,
...
+                                                       tensor_component.end()});
+          } else {
+            components.emplace_back(component_name,
+                                    std::move(tensor_component));
+          }
+        };
+
+    const auto record_tensor_components_impl =
+        [&record_tensor_component_impl, &interpolant](
+            const auto& tensor, const FloatingPointType floating_point_type,
```
Comment (nilsdeppe) [inline src/ParallelAlgorithms/Events/ObserveFields.hpp:253]: This is not meant to be a late review or anything.

Just an FYI, in C++20 you can now do
```cpp
[&record_tensor_component_impl, &interpolant]<typename TensorType>(const TensorType&, ...)
```
to get template parameters in lambdas. You could also do
```cpp
[&record_tensor_component_impl, &interpolant]<typename X, typename Y, typename Z>(const Tensor<X,Y,Z>&, ...)
```
with better names than X,Y,Z :)

**Pull request merged**